pkg/settlement/pseudosettle/pseudosettle.go

```diff
@@ -106,17 +107,22 @@ func (s *Service) handler(ctx context.Context, p p2p.Peer, stream p2p.Stream) (e
 		return err
 	}
 
-	return s.notifyPaymentFunc(p.Address, new(big.Int).SetUint64(req.Amount))
+	return s.accountingAPI.NotifyPaymentReceived(p.Address, new(big.Int).SetUint64(req.Amount))
 }
 
 // Pay initiates a payment to the given peer
-func (s *Service) Pay(ctx context.Context, peer swarm.Address, amount *big.Int) error {
+func (s *Service) Pay(ctx context.Context, peer swarm.Address, amount *big.Int) {
 	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
 	defer cancel()
-
+	var err error
+	defer func() {
+		if err != nil {
```

Comment:
why only send notification of failed payment?

Reply:
afaik as long as at least one other variable is still undeclared `:=` won't actually redeclare existing ones. this is the same  way of error handling as done in the pushsync package.

the happy case is actually handled in L139. the reason why this is not in the defer is so that we don't need to declare the amount in advance (which is the case here anyway, but it won't be in the future).